src/modules/project/pages/projectView/hooks/useProjectShare.ts

```diff
@@ -0,0 +1,83 @@
+import { useState } from 'react'
+
+import { useAuthContext } from '../../../../../context'
+import { copyTextToClipboard } from '../../../../../utils'
+import { useProjectContext } from '../../../context'
+
+enum CampaignSource {
+  /** For content shared by creator */
+  creator = 'creator',
+  /** For content shared by creator */
```

Comment:
All comments say "shared by creator". I'd also suggest adding a description of what each means.